web/packages/core/src/ruffle-player.ts

```diff
@@ -887,8 +903,49 @@ export class RufflePlayer extends HTMLElement {
         return items;
     }
 
-    private showContextMenu(e: MouseEvent): void {
-        e.preventDefault();
+    private clearLongPressTimer(): void {
+        if (this.longPressTimer) {
+            clearTimeout(this.longPressTimer);
+            this.longPressTimer = null;
+        }
+    }
+
+    private startLongPressTimer(): void {
+        const longPressTimeout = 800;
+        this.clearLongPressTimer();
+        this.longPressTimer = setTimeout(
+            () => this.clearLongPressTimer(),
+            longPressTimeout
+        );
+    }
+
+    private checkLongPress(event: PointerEvent): void {
+        if (this.longPressTimer) {
+            this.clearLongPressTimer();
+        } else if (
+            !this.contextMenuSupported &&
+            event.pointerType !== "mouse"
```

Comment:
nit: Do we can to check for `event.pointerType === "touch" || event.pointerType === "pen"` like we do in `pointerDown()`?

Reply:
We could, though this check is arguably more inclusive of all possibilities: https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pointerType#value:

> If the device type cannot be detected by the browser, the value can be an empty string (`""`). If the browser supports pointer device types other than those listed above, the value should be *vendor-prefixed* to avoid conflicting names for different types of devices.

I don't actually own or know of any devices with other pointer types, but they are documented.

Reply:
So with that in mind, I can change this, change pointerDown to be consistent with this instead, or change nothing. What do you think is best?

Reply:
Do we even need this condition at all? I can't think of a scenario where `!this.contextMenuSupported` and `event.pointerType` is not "touch".
Unless I miss something, I think it can be removed.

Reply:
From a theoretical perspective, you're correct. However, according to @n0samu on Discord: "the 'feature detection' method doesn't work unless you actually right-click" and "the solution to this problem is simple, just don't trigger the context menu on pointerup if the type of pointer is mouse". Perhaps a more accurate variabloe name would be `contextMenuTriggered` and perhaps this could use a comment.

Reply:
That's because `this.contextMenuSupported` is set whenthe contextmenu event fires, but this check occurs on `pointerup`, which can fire either before or after the `contextmenu` event on different devices (linux and Windows differ here).

Reply:
Got it. So let's keep the code as-is and add the comment you wrote.